app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardClipboard.java

```diff
@@ -116,11 +116,28 @@ protected boolean handleSelectionExpending(
         if (mArrowSelectionState && ic != null) {
             switch (keyEventKeyCode) {
                 case KeyEvent.KEYCODE_DPAD_LEFT:
-                    ic.setSelection(
-                            Math.max(0, globalSelectionStartPosition - 1), globalCursorPosition);
+                    // A Unicode codepoint can be made up of two Java chars.
+                    // We check if that's what happening before the cursor:
+                    final String toLeft = ic.getTextBeforeCursor(8, 0).toString();
```

Comment:
Why `8`? Is this the longest possible multi-codepoints?
In any case, can you extract that to a const? `protected static final int MAX_MULTI_CODEPOINTS_COUNT = 8;` and use that anywhere you used `8`?

Reply:
8 is a senseless safeguard. A codepoint can either fill up 1 or 2 `char`, and judging from my research, 2 will be the maximum for the foreseeable future.

(When they "run out" of codepoints they'll need to tweak UTF-16, and there's no telling on how would they do it. I also expect that at that point Java will add more methods to treat codepoints in an abstract way, without treating the text as UTF-16.)

Constant extracted and set to 2 on commit 71a8b2ff4b34ce01a4b53be6bf245bf5f7ed28db.